subsys/bluetooth/mesh/light_xyl_cli.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/mesh/light_xyl_cli.h>
+#include "model_utils.h"
+
+static void status_decode(struct bt_mesh_light_xyl_cli *cli,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf, uint32_t opcode,
+			  struct bt_mesh_light_xyl_status *status)
+{
+	if (buf->len != BT_MESH_LIGHT_XYL_MSG_MINLEN_STATUS &&
+	    buf->len != BT_MESH_LIGHT_XYL_MSG_MAXLEN_STATUS) {
+		return;
+	}
+
+	status->params.lightness = net_buf_simple_pull_le16(buf);
+	status->params.x = net_buf_simple_pull_le16(buf);
+	status->params.y = net_buf_simple_pull_le16(buf);
+	status->remaining_time =
+		(buf->len == 1) ?
+			model_transition_decode(net_buf_simple_pull_u8(buf)) :
+			0;
+
+	if (model_ack_match(&cli->ack_ctx, opcode, ctx)) {
+		struct bt_mesh_light_xyl_status *rsp =
+			(struct bt_mesh_light_xyl_status *)
+				cli->ack_ctx.user_data;
+		*rsp = *status;
+		model_ack_rx(&cli->ack_ctx);
+	}
+}
+
+static void xyl_status_handle(struct bt_mesh_model *model,
+			      struct bt_mesh_msg_ctx *ctx,
+			      struct net_buf_simple *buf)
+{
+	struct bt_mesh_light_xyl_cli *cli = model->user_data;
+	struct bt_mesh_light_xyl_status status;
+
+	status_decode(cli, ctx, buf, BT_MESH_LIGHT_XYL_STATUS, &status);
+
+	if (cli->handlers->xyl_status) {
+		cli->handlers->xyl_status(cli, ctx, &status);
+	}
+}
+
+static void target_status_handle(struct bt_mesh_model *model,
+				 struct bt_mesh_msg_ctx *ctx,
+				 struct net_buf_simple *buf)
+{
+	struct bt_mesh_light_xyl_cli *cli = model->user_data;
+	struct bt_mesh_light_xyl_status status;
+
+	status_decode(cli, ctx, buf, BT_MESH_LIGHT_XYL_TARGET_STATUS, &status);
+
+	if (cli->handlers->target_status) {
```

Comment:
```suggestion
	if (cli->handlers && cli->handlers->target_status) {
```
Also, add checking the handler pointer (https://github.com/nrfconnect/sdk-nrf/pull/3273#discussion_r515975821)